### PR TITLE
vdirsyncer: add launchd plist file

### PR DESCRIPTION
**Pull request by benjaminfrank**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Does your submission pass `brew audit --strict --online <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Have you built your formula locally prior to submission with `brew install <formula>`?

### CODE REVIEW

**Comment by tdsmith** [comment]
Thanks for the PR! This looks probably great but I don't know how to review plists; anyone have any comments here?

File: Formula/vdirsyncer.rb [hunk 1]
```diff
@@ -74,6 +74,45 @@ def install
     bin.env_script_all_files(libexec/"bin", :PYTHONPATH => ENV["PYTHONPATH"])
   end
 
+  def caveats
+    "The vdirsyncer launch script defaults to running every 60s with en_US.UTF-8 locale."
```
Comment (dunn) [inline Formula/vdirsyncer.rb]: Is this Homebrew-specific information?

Comment (benjaminfrank) [inline Formula/vdirsyncer.rb]: Yes, as I wrote this plist and chose the params based on what I figured is fitting. And that I could share this with others. Not sure this is how it is supposed to be in homebrew though. I could ask upstream if they want to provide it.

Comment (UniqMartin) [inline Formula/vdirsyncer.rb]: Would be great if you could talk to upstream whether they want to include a `plist` file or what values they recommend for the parameters. Thanks!

**Comment by benjaminfrank** [comment]
Closing this PR as inclusion is pending upstream

**Comment by benjaminfrank** [comment]
I quick question regarding including this plist: some keys are very hombrew specific, e.g. where binaries are installed. How should I deal with this once an example plist is included upstream? I'd probably replace half of the values anyway....

**Comment by UniqMartin** [comment]
> some keys are very hombrew specific, e.g. where binaries are installed. How should I deal with this once an example plist is included upstream?

I think the installation directory is about the only thing that should be variable, but is not specific to Homebrew. The path in the `.plist` file should be changed no matter if the package is installed via Homebrew, some other package manager on OS X, or built from “source”. What else comes to mind?

I think the canonical way would be that the package includes a template of the `.plist` file that contains some placeholder like `@@PREFIX@@` and that gets substituted during installation with the actual path.

**Comment by benjaminfrank** [comment]
I am somewhat new to the whole launchd thing but those come to my mind:
```
<key>Label</key>
<key>WorkingDirectory</key>
<key>ProgramArguments</key>
<key>StandardErrorPath</key>
<key>StandardOutPath</key>
```
I already mentioned the placeholder for the upstream plist, e.g. more a blueprint on how to write a plist than having a concrete one. is the @@PREFIX@@ something homebrew specific or do I need to do a string-replace?

**Comment by UniqMartin** [comment]
> Is the `@@PREFIX@@` something homebrew specific or do I need to do a string-replace?

No, that's not specific to Homebrew. Using placeholders like the suggested one is quite common in build systems of software, where script/configuration/… files need to be installed and need to contain information that is not available prior to building the software.

`vdirsyncer` could distribute a `.plist` file that contains placeholder(s) like the one I suggested and then the Homebrew formula would take care of making necessary substitutions via string replacement. You could try and grep in existing formulae. I'm sure some of them do something very similar, but I can't name one off the top of my head.

**Pull request closed without merging**